Sources/swift-format/Subcommands/LintFormatOptions.swift

```diff
@@ -109,3 +124,18 @@ struct LintFormatOptions: ParsableArguments {
     }
   }
 }
+
+extension [Range<Int>] : @retroactive ExpressibleByArgument {
+  public init?(argument: String) {
+    let pairs = argument.components(separatedBy: ",")
+    let ranges: [Range<Int>] = pairs.compactMap {
+      let pair = $0.components(separatedBy: ":")
+      if pair.count == 2, let start = Int(pair[0]), let end = Int(pair[1]), start <= end {
+        return start ..< end
+      } else {
+        return nil
+      }
+    }
+    self = ranges
+  }
+}
```

Comment:
`@retroactive` won’t compile with a Swift 5.10 compiler and swift-format should be able to build with older compilers. The pattern I usually went with, is


```suggestion

extension [Range<Int>] {
  public init?(argument: String) {
    let pairs = argument.components(separatedBy: ",")
    let ranges: [Range<Int>] = pairs.compactMap {
      let pair = $0.components(separatedBy: ":")
      if pair.count == 2, let start = Int(pair[0]), let end = Int(pair[1]), start <= end {
        return start ..< end
      } else {
        return nil
      }
    }
    self = ranges
  }
}

#if compiler(>=6)
extension : @retroactive ExpressibleByArgument {}
#else
extension : ExpressibleByArgument {}
#endif
```